external/fv3fit/fv3fit/pytorch/training_loop.py

```diff
@@ -32,42 +24,78 @@ class TrainingLoopConfig:
 
     n_epoch: int = 20
     buffer_size: int = 50_000
+    samples_per_batch: int = 1
     save_path: str = "weight.pt"
     multistep: int = 1
 
     def fit_loop(
-        self, loss_config, train_model, train_data, validation_data, optimizer
+        self,
+        train_model: torch.nn.Module,
+        train_data: tf.data.Dataset,
+        validation_data: tf.data.Dataset,
+        optimizer: torch.optim.Optimizer,
+        loss_config: LossConfig,
     ) -> None:
         """
         Args:
             train_model: pytorch model to train
-            train_data: training dataset containing samples to be passed to the model
-            validation_data: validation dataset (one time step prediction)
+            train_data: training dataset containing samples to be passed to the model,
+                should have dimensions [sample, time, tile, x, y, z]
+            validation_data: validation dataset containing samples to be passed
+                to the model, should have dimensions [sample, time, tile, x, y, z]
             optimizer: type of optimizer for the model
-            get_loss: Multistep loss function
-            multistep: number of multi-step loss calculation
+            loss_config: configuration of loss function
         """
-        train_data.shuffle(buffer_size=self.buffer_size)
+        train_data = (
+            train_data.unbatch()
+            .shuffle(buffer_size=self.buffer_size)
+            .batch(self.samples_per_batch)
+        )
         train_data = tfds.as_numpy(train_data)
-        validation_data = tfds.as_numpy(validation_data)
-        device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        validation_data = validation_data.unbatch()
+        n_validation = sequence_size(validation_data)
+        validation_state = (
```

Comment:
Probably worth adding to the docstring that the entirety of the validation data is loaded into memory.

Reply:
If you think this needs to be documented, I'd rather modify the code so it works out-of-memory on the tfdataset. I'll make that change.